api/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -536,25 +536,31 @@ private Builder() {}
 
     /**
      * Sets whether the method is idempotent.  If true, calling this method more than once doesn't
-     * have additional side effects.
+     * have additional side effects. If {@code false}, method is also not safe.
      *
      * @since 1.1.0
      */
     @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1775")
     public Builder<ReqT, RespT> setIdempotent(boolean idempotent) {
       this.idempotent = idempotent;
+      if (!idempotent) {
+        this.safe = false;
+      }
       return this;
     }
 
     /**
      * Sets whether this method is safe.  If true, calling this method any number of times doesn't
-     * have side effects.
+     * have side effects. If {@code true}, method is also idempotent.
      *
      * @since 1.1.0
      */
     @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1775")
     public Builder<ReqT, RespT> setSafe(boolean safe) {
       this.safe = safe;
+      if (safe) {
```

Comment:
This is a little errorprone.  Setting the method to be safe, follow by setting it to be not safe has a side effect of leaving it idempotent.  This behavior should be called out in the docs.

Reply:
Created https://github.com/grpc/grpc-java/pull/6290